### DISCUSSION TITLE
META-INF/morphia-config.properties file not found

### DISCUSSION THREAD

**Discussion by LuuckA21** [MorphiaOrg/morphia#2530] [General]
**Describe the bug**
Configuration file not found, see error:

`
dev.morphia.mapping.MappingException: The configuration file META-INF/morphia-config.properties could not be found.
	at dev.morphia.config.MorphiaConfigHelper.loadConfig(MorphiaConfigHelper.java:105) ~[?:?]
	at dev.morphia.config.MorphiaConfigHelper.loadConfig(MorphiaConfigHelper.java:92) ~[?:?]
	at dev.morphia.Morphia.createDatastore(Morphia.java:93) ~[?:?]
	at me.luucka.moria.DBManager.<init>(DBManager.java:21) ~[moria-0.0.1.jar:?]
	at me.luucka.moria.MoriaPlugin.onEnable(MoriaPlugin.java:9) ~[moria-0.0.1.jar:?]
	at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:281) ~[paper-api-1.20.1-R0.1-SNAPSHOT.jar:?]
	at io.papermc.paper.plugin.manager.PaperPluginInstanceManager.enablePlugin(PaperPluginInstanceManager.java:189) ~[paper-1.20.1.jar:git-Paper-164]
	at io.papermc.paper.plugin.manager.PaperPluginManagerImpl.enablePlugin(PaperPluginManagerImpl.java:104) ~[paper-1.20.1.jar:git-Paper-164]
	at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:507) ~[paper-api-1.20.1-R0.1-SNAPSHOT.jar:?]
	at org.bukkit.craftbukkit.v1_20_R1.CraftServer.enablePlugin(CraftServer.java:640) ~[paper-1.20.1.jar:git-Paper-164]
	at org.bukkit.craftbukkit.v1_20_R1.CraftServer.enablePlugins(CraftServer.java:551) ~[paper-1.20.1.jar:git-Paper-164]
	at net.minecraft.server.MinecraftServer.loadWorld0(MinecraftServer.java:636) ~[paper-1.20.1.jar:git-Paper-164]
	at net.minecraft.server.MinecraftServer.loadLevel(MinecraftServer.java:435) ~[paper-1.20.1.jar:git-Paper-164]
	at net.minecraft.server.dedicated.DedicatedServer.initServer(DedicatedServer.java:308) ~[paper-1.20.1.jar:git-Paper-164]
	at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1101) ~[paper-1.20.1.jar:git-Paper-164]
	at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:318) ~[paper-1.20.1.jar:git-Paper-164]
	at java.lang.Thread.run(Thread.java:833) ~[?:?]
`

** Please complete the following information: **
 - Server Version: 6.0.6
 - Driver Version: 4.10.2
 - Morphia Version: 2.4.2

**Comment by evanchooly** [comment]
do you have that file in `src/main/resources/META-INF/morphia.properties` ?

**Comment by LuuckA21** [comment]
Yes I have
![Screenshot 2023-09-05 224144](https://github.com/MorphiaOrg/morphia/assets/50712503/7cc5136f-c67a-47b2-be51-4ddb5992cc6f)

**Comment by evanchooly** [comment]
If you're using maven or gradle, that file should get copied to your build dir (which is what's actually on the classpath at runtime).  Assuming maven, does `target/classes/META-INF/morphia-config.properties` exist?

**Comment by LuuckA21** [comment]
Yes exist
![Screenshot 2023-09-06 213335](https://github.com/MorphiaOrg/morphia/assets/50712503/f23f80d7-0576-4ebe-a882-de8ad57a09a6)
Same thing in jar file

**Comment by evanchooly** [comment]
There must be something off in your environment because that looks right to me.  
If you could put together a [reproducer](/morphiaorg/reproducer) I'd be happy to take a look and see.

**Comment by LuuckA21** [comment]
I don't understand what I have to do with reproducer

**Comment by evanchooly** [comment]
Create a new project using that as  template (there's a big green button for that) and then recreate the problem in that new project.  It uses maven so if you don't you'll want to recreate your environment as best you can.

**Comment by NOTiFYcouk** [comment]
I've migrated from:

`morphia-core-2.4.2.jar -> morphia-core-3.0.0-20230725.035612-1.jar
`

Configuration: META-INF/morphia-config.properties 

See:  https://morphia.dev/morphia/3.0/configuration.html

Deployed: Sample config files Complete with defaults:

`NOTiFYmoto/src/main/java/META-INF/morphia-config.properties
`

Using Gradle 8.3: JAR:

`META-INF/morphia-config.properties`

MongoDB 7.0.0 with:

```
morphia-core-3.0.0-20230725.035612-1.jar
bson-4.10.2.jar
mongodb-driver-core-4.10.2.jar
mongodb-driver-sync-4.10.2.jar
```

@LuuckA21 It all works fine for me

You say your using: Morphia Version: 2.4.2

Says: "Configuration - Starting in **Morphia 3.0**, configuration will be driven via a configuration file rather than programmatic configuration."

**Comment by evanchooly** [comment]
Thanks for confirming, @NOTiFYcouk  it should be `src/main/resources/` for the `META-INF/` folder but because works "accidentally" because of how maven/gradle work internally.  Does this help, @LuuckA21 ?

**Comment by LuuckA21** [comment]
I upload my project to github
[Project](https://github.com/LuuckA21/Moria)

I have update morphia-core to 2.4.3 but give me this warning

[20:31:16 WARN]: [dev.morphia.config.MorphiaConfig] The configuration file META-INF/morphia-config.properties could not be found. Using default values;

**Reply by LuuckA21** [reply]
I try to move the file but nothing change

**Comment by LuuckA21** [comment]
Still not working @evanchooly

**Comment by LuuckA21** [comment]
I try

2.4.2
2.4.3
3.0.0-snapshot

but still give

dev.morphia.mapping.MappingException: The configuration file META-INF/morphia-config.properties could not be found.

**Comment by LuuckA21** [comment]
I put META-INF/morphia-config.properties  under resources folder and try to put under java folder but nothing change

**Reply by evanchooly** [reply]
Can you try building a reproducer?

**Reply by LuuckA21** [reply]
https://github.com/LuuckA21/Moria

**Reply by LuuckA21** [reply]
My project is a Minecraft plugin and when I start my server it download library file jar from mavencentral
I don't need to shade morphia jar into my jar file

**Reply by evanchooly** [reply]
This isn't runnable.  I don't have a minecraft environment to run this in.  can you recreate the error with a unit test I could run?

**Comment by LuuckA21** [comment]
Testing using reproduce is no sense, I need to use morphia in minecraft plugin and still not working

Reproducer:
https://github.com/LuuckA21/morphia-test

**Comment by LuuckA21** [comment]
I try with reproducer and works well, but this not help me. I put the same code used in reproducer in my minecraft plugin but still not working.

**Reply by NOTiFYcouk** [reply]
Can you connect to your MongoDB collections you created with JavaScript, Then using say MongoDB Compass or Studio 3T to browse your MongoDB data?

**Reply by LuuckA21** [reply]
Im using java and I use mongodbcompass

**Reply by LuuckA21** [reply]
Java: openjdk version "17.0.8.1" 2023-08-22 LTS
Morphia-core: 2.4.4
MongoDB: v7.0.0

**Comment by LuuckA21** [comment]
@evanchooly I downgrade to v2.3.8 because v2.4.x and v3.x snapshot with config file still not working, and after two weeks of discussion nothing has changed

**Comment by evanchooly** [comment]
Everything looks right to me but without a way to reproduce locally there's not much else i can do.

**Reply by LuuckA21** [reply]
It's simple to run a minecraft server and you don't need to have minecraft purchased

**Reply by evanchooly** [reply]
> It's simple to run a minecraft server and you don't need to have minecraft purchased

It's simple if you know how but I'm traveling thelis week for a conference so I don't have that kind of time.  I'd confirm the jar you're giving to the MC server has that file in it.

**Reply by evanchooly** [reply]
> Are you sure Mongo DB "morphia_repro" creates/exists? Can't you avoid using "BottleRocketTest"? Isn't it for AWS?   

No bottelrocket doesn't have anything to do with AWS. It was just a tool I wrote before testcontainers was a thing to be able to test multiple dB server versions.
